document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/mapping/mappers/recentfilinghistory/ApiToRecentFilingHistoryMapper.java

```diff
@@ -67,6 +77,17 @@ private DateTimeFormatter getFormatter() {
 
     private String populateParameters(Object description, Map<String, Object> parameters) {
         StrSubstitutor sub = new StrSubstitutor(parameters, "{", "}");
+
+        for (String parameterKey : parameters.keySet()) {
+
+            if (parameterKey.contains("_date")) {
+
+                LocalDate localDate = LocalDate.parse(parameters.get(parameterKey).toString());
+
+                parameters.replace(parameterKey, localDate.format(getFormatter()));
```

Comment:
Should probably use replace(K key ,V oldValue, V newValue) in order to check that the key and value are the same as the old one. before updating to the new value. As we could potentially have two of the same keys and we update the wrong one.